pkg/river/parser/internal.go

```diff
@@ -683,34 +683,28 @@ func (p *parser) parseFieldList(until token.Token) []*ast.ObjectField {
 //
 //    Field = ( string | identifier ) "=" Expression
 func (p *parser) parseField() *ast.ObjectField {
-	if p.tok != token.STRING && p.tok != token.IDENT {
-		p.addErrorf("expected field name (string or identifier), got %s", p.tok)
-		p.advanceAny(fieldStarter)
-		return nil
-	}
+	var field ast.ObjectField
 
-	field := &ast.ObjectField{
-		Name: &ast.Ident{
+	if p.tok == token.STRING || p.tok == token.IDENT {
+		field.Name = &ast.Ident{
 			Name:    p.lit,
 			NamePos: p.pos,
-		},
-	}
-	if p.tok == token.STRING && len(p.lit) > 2 {
-		// The field name a string literal; unwrap the quotes.
-		field.Name.Name = p.lit[1 : len(p.lit)-1]
-		field.Quoted = true
+		}
+		if p.tok == token.STRING && len(p.lit) > 2 {
+			// The field name a string literal; unwrap the quotes.
```

Comment:
Found a typo in the comment that got copied over :)
```suggestion
			// The field name is a string literal; unwrap the quotes.
```